lldb/source/Plugins/LanguageRuntime/CPlusPlus/CPPLanguageRuntime.cpp

```diff
@@ -90,9 +79,26 @@ class LibCXXFrameRecognizer : public StackFrameRecognizer {
     if (!sc.function)
       return {};
 
+    // Check if we have a regex match
+    bool matches_regex = false;
     for (RegularExpression &r : m_hidden_regex)
-      if (r.Execute(sc.function->GetNameNoArguments()))
+      if (r.Execute(sc.function->GetNameNoArguments())) {
+        matches_regex = true;
+        break;
+      }
+
+    if (matches_regex) {
+      // Only hide this frame if the immediate caller is also within libc++.
+      lldb::StackFrameSP parent_frame =
```

Comment:
`parent_frame` -> `parent_frame_sp`.

Also, we need to check `parent_frame` and `GetThread()` for `nullptr` here